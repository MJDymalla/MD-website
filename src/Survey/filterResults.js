function filterResults(results) {
  var filtered_results = {
    motivation: { intrinsic: 0, introjected: 0, identified: 0, external_motivation: 0 },
    passion: { inventing: 0, founding: 0, developing: 0 },
    effort: { immediate: 0, beyond: 0, develop: 0 },
    growth: 0,
    orientation: { risk_taking: 0, innovativeness: 0, proactivity: 0 },
    self_efficacy: 0,
    need_to_achieve: 0,
    grit: 0,
    ambiguity_tolerance: 0,
    resilience: 0,
    satisfaction: 0
  }

  for (let [key, value] of Object.entries(results)) {
    if (key.includes("Business") || key.includes("Educational")) {
      console.log(key + " - " + value);
    }
    if (key.includes("motivation")) {
      for (let [a, v] of Object.entries(value)) {
        if (a.includes("intrinsic")) {
          filtered_results.motivation.intrinsic += parseInt(v, 10);
        } else if (a.includes("identified")) {
          filtered_results.motivation.identified += parseInt(v, 10);
        } else if (a.includes("introjected")) {
          filtered_results.motivation.introjected += parseInt(v, 10);
        } else if (a.includes("external")) {
          filtered_results.motivation.external_motivation += parseInt(v, 10);
        }
      }
    }
    if (key.includes("passion")) {
      for (let [a, v] of Object.entries(value)) {
        if (a.includes("inventing")) {
          filtered_results.passion.inventing += parseInt(v, 10);
        } else if (a.includes("founding")) {
          filtered_results.passion.founding += parseInt(v, 10);
        } else if (a.includes("developing")) {
          filtered_results.passion.developing += parseInt(v, 10);
        }
      }
    }
    if (key.includes("effort")) {
      for (let [a, v] of Object.entries(value)) {
        if (a === "immediate") {
          filtered_results.effort.immediate += parseInt(v, 10);
        } else if (a === "beyond") {
          filtered_results.effort.beyond += parseInt(v, 10);
        } else if (a === "developing") {
          filtered_results.effort.develop += parseInt(v, 10);
        }
      }
    }
    if (key.includes("growth")) {
      filtered_results.growth += parseInt(value, 10);
    }
    if (key.includes("orientation")) {
      for (let [a, v] of Object.entries(value)) {
        if (a.includes("risk-taking")) {
          filtered_results.orientation.risk_taking += parseInt(v, 10);
        } else if (a.includes("innovativeness")) {
          filtered_results.orientation.innovativeness += parseInt(v, 10);
        } else if (a.includes("proactivity")) {
          filtered_results.orientation.proactivity += parseInt(v, 10);
        }
      }
    }
    if (key.includes("efficacy")) {
      let vals = Object.values(value);
      for (let val of vals) {
        filtered_results.self_efficacy += parseInt(val, 10);
      }
    }
    if (key.includes("need to achieve")) {
      let vals = Object.values(value);
      for (let val of vals) {
        filtered_results.need_to_achieve += parseInt(val, 10);
      }
    }
    if (key.includes("grit")) {
      let vals = Object.values(value);
      for (let val of vals) {
        filtered_results.grit += parseInt(val, 10);
      }
    }
    if (key.includes("tolerance")) {
      let vals = Object.values(value);
      for (let val of vals) {
        filtered_results.ambiguity_tolerance += parseInt(val, 10);
      }
    }
    if (key.includes("resilience")) {
      let vals = Object.values(value);
      for (let val of vals) {
        filtered_results.resilience += parseInt(val, 10);
      }
    }
    if (key.includes("satisfaction")) {
      let vals = Object.values(value);
      for (let val of vals) {
        filtered_results.satisfaction += parseInt(val, 10);
      }
    }
  }
  return filtered_results;
}
export default filterResults;