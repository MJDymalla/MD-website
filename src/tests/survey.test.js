import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-shallow-renderer'; 
import Survey from '../Routes/Survey';

test('should render header correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Survey />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
})



