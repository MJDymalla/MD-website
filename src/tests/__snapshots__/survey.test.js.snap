// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <HeaderBox />
  <div
    className="origin"
  >
    <div
      className="center header-position"
    >
      <i
        className="medium material-icons icon-color"
      >
        blur_on
      </i>
      <header
        className="pageHeaders white-text"
      >
        Assess your mindset
      </header>
    </div>
    <div
      className="container center floating-container z-depth-2"
    >
      <div
        className="floating-content flow-text"
      >
        <div
          className="grey-text text-darken-2 visualRow"
        >
          Clicking the button below will begin the mindset profile.
          <br />
          You will be presented with a series of questions to assess the way you think.
          <br />
          In order to achieve the best results we ask that you ensure each question is answered, and answered truthfully.
        </div>
        <i
          className="small material-icons icon-color"
        >
          linear_scale
        </i>
        <div
          className="grey-text text-darken-1"
        >
          <i>
            We feel it is important to let you know that your answers will not be shown to anyone.
          </i>
        </div>
        <hr
          className="style"
        />
        <p>
          <ForwardRef
            to="/questions"
          >
            <a
              className="btn-small waves-effect waves-light grey darken-4"
              href="#!"
            >
              BEGIN
            </a>
          </ForwardRef>
        </p>
      </div>
    </div>
  </div>
</div>
`;
